react-project/src/components/SimulationModeModal.jsx

```diff
@@ -129,35 +143,25 @@ function SimulationModeModal(props) {
         <button
           type="button"
           className="btn btn-primary"
-          data-bs-toggle="modal"
-          data-bs-target="#exampleModal">
+          onClick={handleShow}
+        >
           Configure Simulation
         </button>
       </div>
 
       {/* Modal */}
-      <div
-        className="modal fade"
-        id="exampleModal"
-        tabIndex="-1"
-        aria-labelledby="exampleModalLabel"
-        aria-hidden="true">
-        <div className="modal-dialog modal-lg">
-          <div className="modal-content">
-            <div className="modal-header">
-              <h1
-                className="modal-title fs-5"
-                id="exampleModalLabel">
-                Simulation Mode Configuration
-              </h1>
-              <button
-                type="button"
-                className="btn-close"
-                data-bs-dismiss="modal"
-                aria-label="Close"></button>
-            </div>
-            <div className="modal-body">
-              <div className="col-auto">
+      <Modal show={show} onHide={handleClose} size="lg">
```

Comment:
Nice, why was I forcing myself to use manual markup for this long, we'll never know